common/config/rush/.pnpmfile.cjs

```diff
@@ -36,7 +36,7 @@ const forcedDep = {};
   [["@itwin/build-tools"], "4.0.0-dev.87"],
   [["electron"], "^23.0.0"],
   [["typedoc"], "0.23.28"],
-  [["typescript"], "^5.0.2"],
+  [["typescript"], "~5.0.2"],
```

Comment:
This doesn't need to be locked anymore btw, latest dev release addresses the ts5.3 issue: https://www.npmjs.com/package/@itwin/eslint-plugin/v/4.0.0-dev.52